pkg/util/tracing/tracer.go

```diff
@@ -646,14 +646,46 @@ func (t *Tracer) ExtractMetaFrom(carrier Carrier) (*SpanMeta, error) {
 	}, nil
 }
 
-// GetActiveSpanFromID retrieves any active span given its span ID.
-func (t *Tracer) GetActiveSpanFromID(spanID uint64) (*Span, bool) {
+// GetActiveRootSpanFromID retrieves any active span given its span ID.
+func (t *Tracer) GetActiveRootSpanFromID(spanID uint64) (*Span, bool) {
 	t.activeSpans.Lock()
 	span, found := t.activeSpans.m[spanID]
 	t.activeSpans.Unlock()
 	return span, found
 }
 
+// SetAllDescendantsSpanVerbosity takes a root Span and sets all its
+// descendant spans' verbosity using setAllChildrenSpanVerbosity. The method
+// returns true if setAllChildrenSpanVerbosity is called successfully and false
+// if not because the root Span was a no-op span.
+func (t *Tracer) SetAllDescendantsSpanVerbosity(rootSpan *Span, to bool) bool {
+	rootCrdbSpan := rootSpan.i.crdb
+	// If we have a no-op root span, we were not successful.
+	if rootCrdbSpan == nil {
+		return false
+	}
+	t.setAllChildrenSpanVerbosity(rootCrdbSpan, to)
+	return true
+}
+
+// SetAllChildrenSpanVerbosity takes a root crdbSpan, sets its children's
+// verbosity if they are not already set appropriately, and makes a recursive
+// call on its children if they have their own children.
+func (t *Tracer) setAllChildrenSpanVerbosity(rootCrdbSpan *crdbSpan, to bool) {
+	rootCrdbSpan.mu.Lock()
+	for _, child := range rootCrdbSpan.mu.recording.children {
+		if to && child.mu.recording.recordingType.load() == RecordingOff {
+			child.enableRecording(nil /* parent */, RecordingVerbose)
+		} else if !to && child.mu.recording.recordingType.load() == RecordingVerbose {
```

Comment:
I wanted to check that the new requested recording type didn't match the existing recording type (see above) and only call `enableRecording` or `disableRecording` if both conditions are true. This would prevent us from having to go through the whole exercise of grabbing and releasing the lock when it's not necessary.

Reply:
See comment above (sorry for using github reviews, reviewable was down).